### ISSUE TITLE
dotnet sln remove -- flattens the solution items

### ISSUE THREAD

**Issue by cveld** [dotnet/cli#10819]
## Steps to reproduce
- Set up an sln with nested solution items.
- Remove a project

## Expected  behavior
The nested solution items should be preserved.

## Actual behavior
The nested solution items get flattened.
For a repro see the following repo: https://github.com/cveld/dotnet-cli-sln-remove-issue

## Environment data
`dotnet --info` output:
.NET Core SDK (reflecting any global.json):
 Version:   2.1.503
 Commit:    4c506e0f35

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 RID:         win81-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.503\

Host (useful for support):
  Version: 2.1.7
  Commit:  cca5d72d48

**Comment by livarcocc** [comment]
@peterhuene just putting this in your backlog.

**Comment by santoshyadav198613** [comment]
Hi @peterhuene,

Can I take this up, first time contributor here, so may need help. 

Regards, 
Santosh

**Comment by peterhuene** [comment]
@santoshyadav198613 we'd be honored.  Feel free to mention me if you run into any issues and I'll help you along the best I can.  Thanks!

**Comment by santoshyadav198613** [comment]
Hi @peterhuene ,

If i understand this issue correctly this is what i have observed:
when we run below command 
dotnet sln WebApplication1.sln remove ConsoleApp1\ConsoleApp1.csproj

this also removes below section 
	GlobalSection(NestedProjects) = preSolution
		{9188D173-EFB7-4EEA-9846-9F5A79F2CB38} = {02AD42F4-9611-4B66-B9B0-BFE409B7D9D6}
	EndGlobalSection
which should not happen , please correct me if i am wrong.

Regards,
Santosh

**Comment by peterhuene** [comment]
That is correct.  It also seems to remove the solution folder `Folder1` for me entirely.

**Comment by santoshyadav198613** [comment]
Ok i have found the code, and built the code locally as well. Will update once resolved.

**Comment by vnwonah** [comment]
@peterhuene @livarcocc I can fix this, however, I find there is a test checking that this particular behavior is exhibited at 

https://github.com/dotnet/cli/blob/3e26f2daf398be5aacd6194c1a06690c656d0866/test/dotnet-sln-remove.Tests/GivenDotnetSlnRemove.cs#L666

Makes me wonder if there's a reason for the behavior or I can go ahead to fix

**Comment by peterhuene** [comment]
Hi @vnwonah.  Thanks for offering to help fix this issue!

I believe the test to be correct for what it tests.  The intention is that if you remove a project and the solution folder containing that project is made empty by the removal, the containing solution folder also gets removed.  If removing the solution folder causes its parent solution folder to now be empty, it gets removed; this process is repeated until the root is reached.

In this repro case, however, the project being removed isn't in a solution folder.  The solution folder hierarchy should remain unchanged after the project is removed.

**Comment by vnwonah** [comment]
Ah, I see that now @peterhuene. Thanks for pointing that out. I’ll open a PR when done.

**Comment by Cosebdd** [comment]
I believe this PR https://github.com/dotnet/cli/pull/12149 also fixes this issue
